packages/html-reporter/src/index.tsx

```diff
@@ -58,8 +58,20 @@ class ZipReport implements LoadedReport {
     const zipURI = await new Promise<string>(resolve => {
       if (window.playwrightReportBase64)
         return resolve(window.playwrightReportBase64);
-      window.addEventListener('message', event => event.source === window.opener && resolve(event.data), { once: true });
-      window.opener.postMessage('ready', '*');
+      if (window.opener) {
+        window.addEventListener('message', event => {
+          if (event.source === window.opener) {
+            localStorage.setItem('playwrightReportBase64', event.data);
```

Comment:
nit: ideally move it out into a constant, something like so its more reasonable in the future.

```suggestion
            localStorage.setItem(kPlaywrightReportStorageForHMR, event.data);
```